lib/Db/RowMapper.php

```diff
@@ -133,295 +327,340 @@ private function resolveSearchValue(string $unresolvedSearchValue, string $userI
-					$filter['columnType'] = $neededColumnTypes[$filter['columnId']];
-					// TODO move resolution for magic fields to service layer
-					if(str_starts_with($filter['value'], '@')) {
-						$filter['value'] = $this->resolveSearchValue($filter['value'], $userId);
-					}
-				}
-			}
-			$qb->andWhere(
-				$qb->expr()->orX(
-					...$this->getFilterGroups($qb, $enrichedFilters)
-				)
-			);
+		// write all changed cells to its db-table
+		foreach ($row->getChangedCells() as $cell) {
+			$this->insertOrUpdateCell($sleeve->getId(), $cell['columnId'], $cell['value']);
 		}
+
+		return $row;
 	}
 
 	/**
-	 * @param int $tableId
-	 * @param int|null $limit
-	 * @param int|null $offset
-	 * @return array
 	 * @throws Exception
 	 */
-	public function findAllByTable(int $tableId, ?int $limit = null, ?int $offset = null): array {
-		$qb = $this->db->getQueryBuilder();
-		$qb->select('t1.*')
-			->from($this->table, 't1')
-			->where($qb->expr()->eq('table_id', $qb->createNamedParameter($tableId)));
-
-		if ($limit !== null) {
-			$qb->setMaxResults($limit);
-		}
-		if ($offset !== null) {
-			$qb->setFirstResult($offset);
-		}
-
-		return $this->findEntities($qb);
+	private function createNewRowSleeve(int $tableId): RowSleeve {
+		$rowSleeve = new RowSleeve();
+		$rowSleeve->setTableId($tableId);
+		$this->updateMetaData($rowSleeve, true);
+		return $this->rowSleeveMapper->insert($rowSleeve);
 	}
 
 	/**
-	 * @param View $view
-	 * @param string $userId
-	 * @param int|null $limit
-	 * @param int|null $offset
-	 * @return array
 	 * @throws Exception
 	 */
-	public function findAllByView(View $view, string $userId, ?int $limit = null, ?int $offset = null): array {
-		$qb = $this->db->getQueryBuilder();
-		$qb->select('t1.*')
-			->from($this->table, 't1')
-			->where($qb->expr()->eq('table_id', $qb->createNamedParameter($view->getTableId(), IQueryBuilder::PARAM_INT)));
-
-
-		$neededColumnIds = $this->getAllColumnIdsFromView($view, $qb);
-		$neededColumnsTypes = $this->columnMapper->getColumnTypes($neededColumnIds);
+	private function createRowSleeveFromExistingData(int $id, int $tableId, string $createdAt, string $createdBy, string $lastEditBy, string $lastEditAt): RowSleeve {
+		$rowSleeve = new RowSleeve();
+		$rowSleeve->setId($id);
+		$rowSleeve->setTableId($tableId);
+		$rowSleeve->setCreatedBy($createdBy);
+		$rowSleeve->setCreatedAt($createdAt);
+		$rowSleeve->setLastEditBy($lastEditBy);
+		$rowSleeve->setLastEditAt($lastEditAt);
+		return $this->rowSleeveMapper->insert($rowSleeve);
+	}
 
-		// Filter
+	/**
+	 * Updates the last_edit_by and last_edit_at data
+	 * optional adds the created_by and created_at data
+	 *
+	 * @param RowSleeve|RowCellSuper $entity
+	 * @param bool $setCreate
+	 * @param string|null $lastEditAt
+	 * @param string|null $lastEditBy
+	 * @return void
+	 */
+	private function updateMetaData($entity, bool $setCreate = false, ?string $lastEditAt = null, ?string $lastEditBy = null): void {
```

Comment:
```suggestion
	private function updateMetaData(Entity $entity, bool $setCreate = false, ?string $lastEditAt = null, ?string $lastEditBy = null): void {
```

or maybe having in interface, so the type can be enforced.